src/Compilers/CSharp/Portable/Parser/LanguageParser_InterpolatedString.cs

```diff
@@ -114,20 +99,22 @@ private ExpressionSyntax ParseInterpolatedStringToken()
                     var interpolation = interpolations[i];
 
                     // Add a token for text preceding the interpolation
-                    var text = Substring(originalText, (i == 0) ? (openQuoteIndex + 1) : (interpolations[i - 1].CloseBracePosition + 1), interpolation.OpenBracePosition - 1);
+                    var text = Substring(originalText,
+                        TextSpan.FromBounds(
+                            i == 0 ? openQuoteIndex + 1 : interpolations[i - 1].CloseBraceSpan.End,
+                            interpolation.OpenBraceSpan.Start));
```

Comment:
instead of needing +1 and -1 on the curly locations, we can actually just use known concepts like .End and .Start to specify what we care about.  (we should probably do this for the openQuote part as well, but i haven't had need to just yet).